src/boot/efi/stub.c

```diff
@@ -898,26 +978,103 @@ static void display_splash(
+                        section_table,
+                        n_section_table,
+                        unified_sections,
+                        /* profile= */ UINT_MAX,
+                        /* validate_base= */ PTR_TO_SIZE(loaded_image->ImageBase),
+                        sections);
+        if (err != EFI_SUCCESS)
+                return log_error_status(err, "Unable to locate embedded base PE sections: %m");
+
+        if (profile != UINT_MAX) {
+                /* And then override them with the per-profile sections of the selected profile */
+                err = pe_locate_profile_sections(
+                                section_table,
+                                n_section_table,
+                                unified_sections,
+                                profile,
+                                /* validate_base= */ PTR_TO_SIZE(loaded_image->ImageBase),
+                                sections);
+                if (err != EFI_SUCCESS && !(err == EFI_NOT_FOUND && profile == 0)) /* the first profile is implied if it doesn't exist */
+                        return log_error_status(err, "Unable to locate embedded per-profile PE sections: %m");
+        }
+
+        if (!PE_SECTION_VECTOR_IS_SET(sections + UNIFIED_SECTION_LINUX))
+                return log_error_status(EFI_NOT_FOUND, "Image lacks .linux section.");
+
+        return EFI_SUCCESS;
+}
+
+static void settle_command_line(
                 EFI_LOADED_IMAGE_PROTOCOL *loaded_image,
                 const PeSectionVector sections[static _UNIFIED_SECTION_MAX],
-                char16_t **ret_cmdline,
+                char16_t **cmdline,
                 int *parameters_measured) {
 
         assert(loaded_image);
         assert(sections);
+        assert(cmdline);
 
-        if (use_load_options(image, loaded_image, /* have_cmdline= */ PE_SECTION_VECTOR_IS_SET(sections + UNIFIED_SECTION_CMDLINE), ret_cmdline)) {
-                /* Let's measure the passed kernel command line into the TPM. Note that this possibly
-                 * duplicates what we already did in the boot menu, if that was already used. However, since
-                 * we want the boot menu to support an EFI binary, and want to this stub to be usable from
-                 * any boot menu, let's measure things anyway. */
-                bool m = false;
-                (void) tpm_log_load_options(*ret_cmdline, &m);
-                combine_measured_flag(parameters_measured, m);
-        } else
-                *ret_cmdline = mangle_stub_cmdline(pe_section_to_str16(loaded_image, sections + UNIFIED_SECTION_CMDLINE));
+        /* This determines which command line to use. On input *cmdline contains the custom passed in cmdline
+         * if there is any.
+         *
+         * We'll suppress the custom cmdline if we are in Secure Boot mode, and if either there is already
+         * a cmdline baked into the UKI or we are in confidential VM mode. */
```

Comment:
I don't really like this. Why pick up the custom command line at all if we're in a secure boot mode? Why not just not pick it up?

Reply:
This is not changed by this patch. This always has been that way.

The model is that if you built in `.cmdline` then it counts in SB mode, no exceptions. But if you leave it out, then anything is allowed.

One might criticize that, but I am pretty sure that's a discussion to have separately, and that should be addressed in a separate PR, since this PR is really not about that and is not changing the behaviour.

Reply:
I think we misunderstand each other. I'm specifically talking about not shortcutting anymore in `process_arguments()` which we did before. That is definitely changing in this PR.

Reply:
I think my reply to your other comment cleans that up?